src/NServiceBus.Core/Pipeline/PipelineExecutionExtensions.cs

```diff
@@ -0,0 +1,91 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using Pipeline;
+
+    static class PipelineExecutionExtensions
+    {
+        // ReSharper disable once SuggestBaseTypeForParameter
+        public static Func<TRootContext, Task> CreatePipelineExecutionFuncFor<TRootContext>(this IBehavior[] behaviors)
+            where TRootContext : IBehaviorContext
+        {
+            return (Func<TRootContext, Task>)behaviors.CreatePipelineExecutionExpression();
+        }
+
+        /// <code>
+        /// rootContext
+        ///    => behavior1.Invoke(rootContext,
+        ///       context1 => behavior2.Invoke(context1,
+        ///        ...
+        ///          context{N} => behavior{N}.Invoke(context{N},
+        ///             context{N+1} => TaskEx.Completed))
+        /// </code>
+        public static Delegate CreatePipelineExecutionExpression(this IBehavior[] behaviors, List<Expression> expressions = null)
+        {
+            Delegate lambdaExpression = null;
+            var length = behaviors.Length - 1;
```

Comment:
nitpicking, but I find the name `length` a bit tricky here (since obviously, it's not the length, but length - 1), so maybe `behaviorCount` or something like that?